apis/v1beta1/collector_webhook_test.go

```diff
@@ -547,7 +548,9 @@ func TestCollectorDefaultingWebhook(t *testing.T) {
 				assert.NoError(t, test.expected.Spec.Config.Service.ApplyDefaults(logr.Discard()), "could not apply defaults")
 			}
 			assert.NoError(t, err)
-			assert.Equal(t, test.expected, test.otelcol)
+			if diff := cmp.Diff(test.expected, test.otelcol); diff != "" {
```

Comment:
I’d like to use cmp.Diff instead of assert.Equal because debugging was difficult. assert.Equal doesn’t show a detailed enough diff and omits some parts if the object is too deep.

Reply:
Can you add this as a comment to the code? I think it makes sense for future readers.